go/host/enclave/guardian.go

```diff
@@ -456,6 +452,15 @@ func (g *Guardian) processL1BlockTransactions(block *common.L1Block) {
 			}
 		}
 	}
+
+	if len(contractAddressTxs) > 0 {
+		go func() {
+			err := g.sl.L1Publisher().ResyncImportantContracts()
+			if err != nil {
+				g.logger.Error("Could not resync important contracts", log.ErrKey, err)
```

Comment:
This kinda breaks the security of the block ingestion, if we go back and do an eth_call someone malicious can interject. Should follow the same pattern of decoding the tx that it's known for a fact is "safe" ?